File: performance/hex-to-binary-performance/hex-to-bin.py

# Read the hex file and convert it to a JPG file
def hex_to_jpg(hex_file_path, jpg_file_path):
   with open(hex_file_path, 'r') as hex_file:
      hex_data = hex_file.read()

   # Convert hex to binary
   binary_data = bytes.fromhex(hex_data)

   # Write binary data to a JPG file
   with open(jpg_file_path, 'wb') as jpg_file:
      jpg_file.write(binary_data)

# Example usage
hex_to_jpg('data/testdata.hex', 'data/python.dat')

File: performance/hex-to-binary-performance/hex-to-bin.pl
#!/usr/bin/env perl

# convert hex string to binary
# usage: hex-to-bin.pl <file>
# this is just a POC script to read a file containing hex characters
# and convert them to binary
# it is necessary to read and convert 2 hex characters at a time

use strict;
use warnings;
use IO::File;

use Data::Dumper;

my $file = shift @ARGV or die "Usage: $0 <file>\n";

my $io = IO::File->new($file, 'r') or die "Cannot open file '$file': $!\n";

# read 1 byte at a time

# read 1 hex character at a time
my $c = '';
while ( $io->read($c,2) ) {
	# convert to binary
	print pack('H*',uc($c));
	#warn "$c";
}

#warn "\n";






